pytorch/3.7-predicting-house-prices.ipynb

```diff
@@ -0,0 +1,656 @@
+    "\n",
+    "We are also monitoring a new metric during training: `mae`. This stands for Mean Absolute Error. It is simply the absolute value of the difference between the predictions and the targets. For instance, a MAE of 0.5 on this problem would mean that our predictions are off by \\$500 on average."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "corresponding-authentication",
+   "metadata": {},
+   "source": [
+    "## Fit with Orca Estimator\n",
+    "We can create the estimator with the following, and we can fit our data using the estimator. Let's try training the network for a bit longer: 500 epochs. To keep a record of how well the model did at each epoch, we will modify our training loop to save the per-epoch validation score log:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "associate-brisbane",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "from zoo.orca.learn.pytorch import Estimator \n",
+    "from zoo.orca.learn.metrics import MAE\n",
+    "from zoo.orca.learn.trigger import EveryEpoch\n",
+    "\n",
+    "k = 4\n",
+    "num_val_samples = len(train_data) // k\n",
+    "num_epochs = 300\n",
+    "all_scores = []\n",
+    "for i in range(k):\n",
+    "    print('processing fold #', i)\n",
+    "    # Prepare the validation data: data from partition # k\n",
+    "    val_data = train_data[i * num_val_samples: (i + 1) * num_val_samples]\n",
+    "    val_targets = train_targets[i * num_val_samples: (i + 1) * num_val_samples]\n",
+    "\n",
+    "    # Prepare the training data: data from all other partitions\n",
+    "    partial_train_data = np.concatenate(\n",
+    "        [train_data[:i * num_val_samples],\n",
+    "         train_data[(i + 1) * num_val_samples:]],\n",
+    "        axis=0)\n",
+    "    partial_train_targets = np.concatenate(\n",
+    "        [train_targets[:i * num_val_samples],\n",
+    "         train_targets[(i + 1) * num_val_samples:]],\n",
+    "        axis=0)\n",
+    "    \n",
+    "    # Build data loader with training and validation data\n",
+    "    train_loader = data_loader_creator(partial_train_data, partial_train_targets, batch_size=16, shuffle=True)\n",
+    "    val_loader = data_loader_creator(val_data, val_targets, batch_size=1, shuffle=False)\n",
+    "\n",
+    "    # Build the Keras model (already compiled)\n",
+    "    model = RegressionNet()\n",
+    "    model.train()\n",
+    "    est = Estimator.from_torch(model=model, optimizer=optimizer, loss=criterion, metrics=[MAE()])\n",
+    "    # Train the model (in silent mode, verbose=0)\n",
+    "    results = []\n",
+    "    for i in range(num_epochs):\n",
+    "        est.fit(data=train_loader, epochs=i+1, checkpoint_trigger=EveryEpoch())\n",
```

Comment:
why epochs is i+1 not 1? Is the implementation of bigdl backend continues the previous epoch?

Reply:
As described above, if I write 1 then the estimator will only train for 1 epoch at the first loop and won't train in the rest of the loops, which is a bit strange...

Reply:
@qiuxin2012 We need to fix this issue. Seems previously in KerasNet we have done this, not sure can migrate to here.

Reply:
@hkvision   This `epoch` means the end epoch of this `fit`, not how many epoch to train. It's right behaviour.